src/platform/qt/GBAKeyEditor.cpp

```diff
@@ -396,14 +396,14 @@ void GBAKeyEditor::updateJoysticks() {
 	m_controller->updateJoysticks();
 	m_controller->recalibrateAxes();
 
+	// Block the currentIndexChanged signal while rearranging the combo box
+	auto wasBlocked = m_profileSelect->blockSignals(true);
 	m_profileSelect->clear();
 	m_profileSelect->addItems(m_controller->connectedGamepads(m_type));
 	int activeGamepad = m_controller->gamepad(m_type);
+	m_profileSelect->setCurrentIndex(activeGamepad);
+	m_profileSelect->blockSignals(wasBlocked);
+
 	selectGamepad(activeGamepad);
-	if (activeGamepad > 0) {
-		m_profileSelect->setCurrentIndex(activeGamepad);
-	}
-	lookupAxes(m_controller->map());
```

Comment:
Can you actually remove these? This is needed to update the axis and hat mapping, so it might get out of date if the mapping changes (e.g. due to switching which controller you're using)